### ISSUE TITLE
Discuss pulling updates from lincc-frameworks template

### ISSUE THREAD

**Issue by camposandro** [LSSTDESC/RAIL-project-template#34]
It would be worth discussing whether this template should be updated with the latest version of the [Python Project Template](https://github.com/lincc-frameworks/python-project-template) or not, as some of the features of the original template have changed. For example, the publish-to-pypi workflow does not require a token anymore and it simplifies package deployment by requesting the registration of a trusted publisher on PyPI.

**Comment by drewoldag** [comment]
My, perhaps short sighted, take on this is that the template here has become rather specialized. We pulled out a lot of the customizability from this template when we initially forked it, and I'm not sure how easy it would be to update it with the latest changes. 

I did update the `publish-to-pypy` action, as well as the linting and daily smoke test to match what is in the LINCC Frameworks template. 

I'm open to counter points, but my thought is that we shouldn't tackle this right now, and reconsider at a later date.
